### PR TITLE
Adds support for /completion and /completion/resolve endpoints for Cake.

### LINKED ISSUE #1939: Missing Cake Response handlers in V1.37.1
Seems version 1.37.1 broke some actions in Cake scripts, such as "/completion". Trying a completion action returns this in debug:
`
[dbug]: OmniSharp.Stdio.Host
        ************  Response ************ 
{
  "Request_seq": 19,
  "Command": "/completion",
  "Running": true,
  "Success": false,
  "Message": "\"System.NotSupportedException: Cake does not support /completion\\r\\n   at OmniSharp.Endpoint.EndpointHandler`2.<HandleRequestForLanguage>d__20.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Host\\\\Endpoint\\\\EndpointHandler.cs:line 233\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Endpoint.EndpointHandler`2.<Process>d__16.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Host\\\\Endpoint\\\\EndpointHandler.cs:line 131\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Stdio.Host.<HandleRequest>d__13.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Stdio\\\\Host.cs:line 215\"",
  "Body": null,
  "Seq": 704,
  "Type": "response"
}
`

Seems like versions below 1.37.1 seem to be unaffected.

After some local debugging, it seems like all it needs is two more CakeRequestHandlers that deal with the specific requests, and reroute them to the C# handlers like so:

```
[OmniSharpHandler(OmniSharpEndpoints.Completion, Constants.LanguageNames.Cake), Shared]
    public class CompletionHandler : CakeRequestHandler<CompletionRequest, CompletionResponse>
    {
        [ImportingConstructor]
        public CompletionHandler(OmniSharpWorkspace workspace)
            : base(workspace)
        {
        }
    }

    [OmniSharpHandler(OmniSharpEndpoints.CompletionResolve, Constants.LanguageNames.Cake), Shared]
    public class CompletionResolveHandler : CakeRequestHandler<CompletionResolveRequest, CompletionResolveResponse>
    {
        [ImportingConstructor]
        public CompletionResolveHandler(OmniSharpWorkspace workspace)
            : base(workspace)
        {
        }
    }

```

But I'm not that familiar with the code base, it's likely some other commands were broken too.

### PR DESCRIPTION
**Pull request by mholo65**
- Fixes #1939

### CODE REVIEW

File: tests/OmniSharp.Cake.Tests/CompletionFacts.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                Assert.StartsWith(
+                    "```csharp\nvoid Information(string format, params object[] args)",
+                    resolved.Item?.Documentation);
+            }
+        }
+
+        // [Fact]
+        // public async Task ShouldGetCompletionWithAdditionalTextEdits()
+        // {
+        //     const string input = @"Regex.Repl$$";
```
Comment (mholo65) [inline tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: Anyone know how to get a completion with Additional Text? In Cake, e.g. we don't have `System.Text.RegularExpressions` namespace imported by default. This was my attempt to get a completion from `CompletionItemExtensions.TypeImportCompletionProvider`. I also tried `var regex == new Rege$$`, but that didn't work.

Comment (filipw) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: it's disabled by default, you need to pass in an option `RoslynExtensionsOptions:EnableImportCompletion`
https://github.com/OmniSharp/omnisharp-roslyn/blob/master/tests/OmniSharp.Roslyn.CSharp.Tests/CompletionFacts.cs#L1399

Comment (mholo65) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: Ah, thank you!

Comment (mholo65) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: Auch! Does AdditionalTextEdits always contain the whole buffer? //cc @333fred 
image

Comment (333fred) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: Unfortunately, yes.

Comment (mholo65) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: > Unfortunately, yes. 

Thanks for the quick reply. In that case I think I'll just remove all additional texts for Cake files. Can Completion return Additional text, or is it just the Resolver that adds this?

Comment (333fred) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs:92]: Override completion can do this as well, and those are not usually so long.

Comment (mholo65) [reply tests/OmniSharp.Cake.Tests/CompletionFacts.cs]: Thanks! I managed to get override completion in at least. Or if it doesn't bring in an unimported type.

**Review by filipw — APPROVED** [review]

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**